job_bundles/README.md

```diff
@@ -0,0 +1,309 @@
+`{{Param.OutputDir}}` will resolve to the [path-mapped location](https://github.com/OpenJobDescription/openjd-specifications/wiki/How-Jobs-Are-Run#path-mapping)
+on the Worker where the file and directory are located.
+After the embedded file has been written
+to disk, the Task's `onRun` command is resolved to `bash <location-of-embeded-file>` and then run.
+
+Please see the Open Job Description specification for additional details on [how Jobs are run](https://github.com/OpenJobDescription/openjd-specifications/wiki/How-Jobs-Are-Run).
+
+Additional samples of Job Templates can be found in all of the Job Bundles in this repository as well as in the 
+[samples provided](https://github.com/OpenJobDescription/openjd-specifications/tree/mainline/samples)
+by Open Job Description. Please see https://github.com/OpenJobDescription/openjd-specifications/wiki
+for more information on Job Templates.
+
+### Elements - Parameter Values
+
+The `parameter_values.yaml`/`parameter_values.json` file in a Job Bundle gives you a place to "bake"
+the values of some of the Job Parameters, and/or [deadline:CreateJob API](https://docs.aws.amazon.com/deadline-cloud/latest/APIReference/API_CreateJob.html)
+request arguments into the Job Bundle so that the values do not have to be given when submitting a Job using the
+Job Bundle. The Job Bundle submission UI that is created will still allow values for these parameters to be modified.
+
+The format of the file, in YAML, is:
+
+```yaml
+parameterValues:
+- name: <string>
+  value: <integer>, <float>, or <string>
+- name: <string>
+  value: <integer, <float>, or <string>
+... repeating as necessary
+```
+
+Each element of the `parameterValues` list in the file must be one of the following: 
+
+1. A Job Parameter defined in the Job Bundle's Job Template; 
+2. A Job Parameter defined in a Queue Environment on the Queue that you are submitting the Job to;
+3. A special parameter that is passed to the [deadline:CreateJob API](https://docs.aws.amazon.com/deadline-cloud/latest/APIReference/API_CreateJob.html)
+   when creating a Job with the Job Bundle:
+    * `deadline:priority` - The value must be an integer, and is passed as the `priority` request parameter to the API.
+    * `deadline:targetTaskRunStatus` - Value must be a string, and is passed as the `targetTaskRunStatus` request parameter to the API.
+    * `deadline:maxFailedTasksCount` - Value must be an integer, and is passed as the `maxFailedTasksCount` request parameter to the API.
+    * `deadline:maxRetriesPerTask` - Value must be an integer, and is passed as the `maxRetriesPerTask` request parameter to the API.
+
+While a Job Template is always a template rather than a specific Job to run, the parameter values file allows
+a Job Bundle to act as a template if some Job Parameters do not have values defined in this file, or as a specific job submission
+if all Job Parameters have values.
```

Comment:
I suppose a job template could also be considered a specific job submission under the same condition (all of the parameters in the template have defaults), right?

What does the parameter values file add beyond using a job template and modifying the defaults? I think the concept we're trying to document here is feature/behavior of the `deadline-cloud` library when it submits a job bundle. It persists a copy of the job bundles with the chosen job parameter values baked into the parameter values file. In this way, the job bundles in the job history directory act as a job submission history. Those bundles can be submitted using the baked parameter values to repeat the job submission.

Reply:
> What does the parameter values file add beyond using a job template and modifying the defaults?

That and providing values for a small handful of the CreateJob API parameters.

 What I'm trying to convey here is that if you have a parameter_values file and that file has values for all non-defaulted job parameters, then the Bundle is, in effect, a representation of a specific concrete Job rather than a template from which Jobs can be created.

And, I think, with that I see a better way to word this paragraph.

Reply:
There was an offline discussion where we discussed that job bundle parameter values do have the same behavior of acting as a default value when submitting a bundle and that they can be overridden.

The differences above and beyond a job template defaults are:
1.  Job bundle parameter values accept wider ranges of values, such as:
    1.  Allow specifying values for `PATH` parameters outside the job bundle directory
    2.  Allow specifying `deadline:*` values used in the `CreateJob` Deadline Cloud API requests.
2. A much more simple way for tooling/automation to provide new/modified default values for a job template: a simple JSON file with name/value pairs

We agreed that this `README.md` is intended to be a high-level informal introduction to job bundles. We can adjust the documentation in the future to help clarify to readers that parameter values can be overridden and are still allow providing different values when submitting a job from the bundle.